app/models/user.rb

```diff
@@ -1045,6 +1045,8 @@ def cannot_organize_competition_reasons
   end
 
   def cannot_edit_data_reason_html(user_to_edit)
+    return I18n.t('users.edit.cannot_edit.reason.no_access') unless user_to_edit == self || can_edit_any_user?
```

Comment:
Where does this new check come from? I can understand from the code what this check does, you don't need to explain that in detail. What I want to know is:
1. Why was this check never necessary before?
2. Will this affect other parts in the code (outside of your current PR) that rely on `cannot_edit_data_reason_html`?

Reply:
So this was used in two places earlier:
1. Getting editable fields of user
2. Show "cannot edit" warning in user edit page

For 1, this check was already there before calling this method. I've changed that as well below.

For 2, it will be good to have this check because the user won't be anyway able to edit as these are not added in the editable fields. So this check will be useful.

This answered your both questions I assume.